= Outline

== Introduction

* Indexing & seeding
* Processing time bottleneck
* Memory bottleneck
* Accelerate read mapping
* Processing near memory
* cf. Proposal

== Background
* k-mer, minimizer
* seeding algorithm
* Filtering
* FPGA & HBM

== Design
=== Reference genome indexing
* Algorithm constraint
** reduce the number of memory accesses
** contiguous memory accesses
** size of the human DNA
** filtering
* Construction of the hash table
* Design constrain
** partitioning for HBM

=== Read encoding
** 4-bit encoding and bit-packing
** Constant length vs Variable length

=== Choice of Parameters
* Size of the buffers
* Adjacency test

=== FPGA design
* Pipeleine
* Data flow
* HBM

== Implementation
=== Methodology
* Readset: Illumina, Pacbio
* CPU implementation for baseline
* Vitis
* Tapasco

== Results
* Performances of the CPU implementation
** Scaling?
** Comparison with minimap2
* Estimation of the performances of the FPGA implementation

== Future work
* kraken 2
* Full design with alignment
* different minimizers

== Conclusion
